### PR TITLE
Add organizer registration and login

### PR DESCRIPTION
**Pull request by CJ4n**

### CODE REVIEW

File: src/main/java/com/team3/central/CentralApplication.java [hunk 1]
```diff
@@ -6,8 +6,8 @@
 @SpringBootApplication
 public class CentralApplication {
 
-	public static void main(String[] args) {
-		SpringApplication.run(CentralApplication.class, args);
-	}
```
Comment (adamdab) [inline src/main/java/com/team3/central/CentralApplication.java]: Please change code styling in IDE to the one I send on discord group or such changes will be created (unnecessary review)

File: src/main/java/com/team3/central/configuration/EmailConfiguration.java [hunk 2]
```diff
@@ -14,40 +13,40 @@
...
+    Properties props = mailSender.getJavaMailProperties();
+    props.put("mail.transport.protocol", "smtp");
+    props.put("mail.smtp.auth", mailServerAuth);
+    props.put("mail.smtp.starttls.enable", mailServerStartTls);
+    props.put("mail.debug", "true");
 
-        return mailSender;
-    }
+    return mailSender;
+  }
```
Comment (adamdab) [inline src/main/java/com/team3/central/configuration/EmailConfiguration.java]: styling (same coment as above)

File: src/main/java/com/team3/central/controllers/OrganizerApiImpl.java [hunk 3]
```diff
@@ -1,9 +1,135 @@
...
+   *
+   * @param id   id of Organizer (required)
+   * @param code code from email (required)
+   * @return account confirmed (status code 201) or code wrong (status code 400)
+   */
+  @Override
+  public ResponseEntity<Organizer> confirm(String id, String code) {
+    var organizerEntityResponseEntity = organizerService.confirm(id, code);
+    if (!organizerEntityResponseEntity.hasBody()) {
+//      organizerEntityResponseEntity.se
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/OrganizerApiImpl.java]: artifact please delete

File: src/main/java/com/team3/central/controllers/OrganizerApiImpl.java [hunk 4]
```diff
@@ -1,9 +1,135 @@
...
+    OrganizerEntity organizerEntity = new OrganizerEntity();
+    organizerEntity.setEmail(organizerDto.getEmail());
+    organizerEntity.setId(organizerDto.getId());
+    organizerEntity.setName(organizerDto.getEmail());
+    organizerEntity.setPassword(organizerDto.getEmail());
+//    organizerEntity.setEvents(organizerDto.getEvents()); TODO: convert entity event to model event
+    organizerEntity.setIsAuthorised(
+        organizerDto.getStatus() == StatusEnum.CONFIRMED);
+    return organizerEntity;
+  }
```
Comment (adamdab) [inline src/main/java/com/team3/central/controllers/OrganizerApiImpl.java]: Create a separate organizer mapper class (add mappers package)

File: src/main/java/com/team3/central/repositories/EventRepository.java [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
-interface EventRepository extends CrudRepository<Event, Long>{
-    Event findById(long id);
-    List<Event> findAll();
-    //byCategory
+interface EventRepository extends CrudRepository<Event, Long> {
+
+  Event findById(long id);
+
+  List<Event> findAll();
+  //byCategory
```
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/EventRepository.java]: Try not to change styling of not necessary, it only makes more noise in review

File: src/main/java/com/team3/central/repositories/ReservationRepository.java [hunk 6]
```diff
@@ -6,5 +6,6 @@
 
 @Repository
 interface ReservationRepository extends CrudRepository<Reservation, Long> {
-    Reservation findById(long id);
+
+  Reservation findById(long id);
```
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/ReservationRepository.java]: styling

File: src/main/java/com/team3/central/repositories/UserRepository.java [hunk 7]
```diff
@@ -6,5 +6,6 @@
 
 @Repository
 interface UserRepository extends CrudRepository<User, Long> {
-    User findById(long id);
+
+  User findById(long id);
```
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/UserRepository.java]: styling

File: src/main/java/com/team3/central/repositories/entities/Category.java [hunk 8]
```diff
@@ -11,26 +12,24 @@
...
-    private Set<Event> events;
+  @ManyToMany(mappedBy = "categories")
+  private Set<Event> events;
 
-    public Category(String name) {
-        this.name = name;
-    }
+  public Category(String name) {
+    this.name = name;
+  }
```
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/entities/Category.java]: styling

File: src/main/java/com/team3/central/repositories/entities/Event.java [hunk 9]
```diff
@@ -14,48 +15,46 @@
...
+  @OneToMany(mappedBy = "event")
+  private Set<Reservation> reservations;
+
+  @ManyToMany
+  @JoinTable(
+      name = "events_categories",
+      joinColumns = @JoinColumn(name = "event_id"),
+      inverseJoinColumns = @JoinColumn(name = "category_id")
+  )
+  private Set<Category> categories;
```
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/entities/Event.java]: styling

File: src/main/java/com/team3/central/repositories/entities/Organizer.java
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/entities/Organizer.java:1]: why it has been deleted/renamed ?

File: src/main/java/com/team3/central/repositories/entities/OrganizerEntity.java
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/entities/OrganizerEntity.java:1]: why it has been added ?

File: src/main/java/com/team3/central/repositories/entities/Reservation.java [hunk 10]
```diff
@@ -17,28 +17,29 @@
...
+  public Reservation(User user, Event event) {
+    this.user = user;
+    this.event = event;
+  }
+
+  public Reservation(User user, Event event, Long placeOnSchema) {
+    this.user = user;
+    this.event = event;
+    this.placeOnSchema = placeOnSchema;
+  }
```
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/entities/Reservation.java]: style

File: src/main/java/com/team3/central/repositories/entities/User.java
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/entities/User.java:1]: style

File: src/main/java/com/team3/central/repositories/entities/enums/EventStatus.java
Comment (adamdab) [inline src/main/java/com/team3/central/repositories/entities/enums/EventStatus.java:1]: style

File: src/main/java/com/team3/central/security/config/ApplicationConfig.java [hunk 11]
```diff
@@ -20,12 +19,22 @@ public class ApplicationConfig {
...
     return new BCryptPasswordEncoder();
   }
+//  public PasswordEncoder passwordEncoder() {
+//    return new BCryptPasswordEncoder();
+//  } why this donst work??
+
+//  @Bean
+//  public ModelMapper modelMapper() {
+//    return new ModelMapper();
+//  }
```
Comment (adamdab) [inline src/main/java/com/team3/central/security/config/ApplicationConfig.java]: if it won't be used delete this please

File: src/main/java/com/team3/central/security/config/SecurityConfig.java
Comment (adamdab) [inline src/main/java/com/team3/central/security/config/SecurityConfig.java:1]: style

File: src/main/java/com/team3/central/services/AuthenticationService.java [hunk 12]
```diff
@@ -2,7 +2,6 @@
 
 import lombok.RequiredArgsConstructor;
 import org.springframework.security.authentication.AuthenticationManager;
-
```
Comment (adamdab) [inline src/main/java/com/team3/central/services/AuthenticationService.java]: style

File: src/main/java/com/team3/central/services/EmailService.java
Comment (adamdab) [inline src/main/java/com/team3/central/services/EmailService.java:1]: style

File: src/test/java/com/team3/central/CentralApplicationTests.java
Comment (adamdab) [inline src/test/java/com/team3/central/CentralApplicationTests.java:1]: style

File: src/test/java/com/team3/central/controllers/CategoriesApiImplTest.java
Comment (adamdab) [inline src/test/java/com/team3/central/controllers/CategoriesApiImplTest.java:1]: style

File: src/test/java/com/team3/central/controllers/EventsApiImplTest.java
Comment (adamdab) [inline src/test/java/com/team3/central/controllers/EventsApiImplTest.java:1]: style

File: src/main/java/com/team3/central/repositories/entities/Organizer.java
Comment (CJ4n) [reply src/main/java/com/team3/central/repositories/entities/Organizer.java:1]: Having two classes with the same name but responsible for different thing was very confusing and required using very long names like com.team3.central.openapi.model.Organizer. I think we should rename all our classes-eneities to adhere to this convention

**Review by adamdab — APPROVED** [review]

**Pull request merged**